plugins/coverage/index.js

```diff
@@ -13,6 +15,62 @@ const tokenRoute = require('./token');
 exports.register = (server, options, next) => {
     const { coveragePlugin } = options;
 
+    /**
+     * Get extra coverage info such as:
+     * - annotations for scope
+     * - jobId and jobName for PR
+     * @method getCoverageConfig
+     * @param {Object}      config              Configuration object
+     * @param {Pipeline}    config.jobId        Screwdriver job ID
+     * @param {Job}         [config.prNum]      PR number
+     * @param {Build}       [config.scope]      Scope (pipeline or job)
+     * @return {Promise}                        Resolves to config object
+     */
+    server.expose('getCoverageConfig', ({ scope, jobId, prNum }) => {
+        const { jobFactory } = server.root.app;
+        const coverageConfig = {
+            jobId,
+            annotations: scope ? { [COVERAGE_SCOPE_ANNOTATION]: scope || null } : {},
+            prNum
+        };
+        const prRegex = /^PR-(\d+)(?::([\w-]+))?$/;
+
+        if (jobId) {
+            // Get job, scope, and PR info
+            return jobFactory.get(jobId).then(job => {
+                if (!job) {
+                    throw boom.notFound(`Job ${jobId} does not exist`);
+                }
+
+                const isPR = job.isPR();
+
+                if (!scope) {
+                    coverageConfig.annotations =
+                        job.permutations[0] && job.permutations[0].annotations ? job.permutations[0].annotations : {};
+                }
+
+                // If scope is job and job is pull request, set jobId and jobName
+                if (coverageConfig.annotations[COVERAGE_SCOPE_ANNOTATION] === 'job' && isPR) {
+                    const prNameMatch = job.name.match(prRegex);
+
+                    coverageConfig.jobId = job.prParentJobId;
+                    coverageConfig.jobName = prNameMatch && prNameMatch.length > 1 ? prNameMatch[2] : job.name;
```

Comment:
is `coverageConfig.jobName` required ? we don't even need to fetch job here, all info is present in `buildCredentials` and can be passed here.

Reply:
yes, we'll need it for `projectName` for `/info` endpoint